File: ftc_app-3.5/TeamCode/src/main/java/org/firstinspires/ftc/teamcode/robots/RelicRecoveryRobot.java
package org.firstinspires.ftc.teamcode.robots;

import com.qualcomm.robotcore.eventloop.opmode.LinearOpMode;
import com.qualcomm.robotcore.hardware.ColorSensor;
import com.qualcomm.robotcore.hardware.DcMotor;
import com.qualcomm.robotcore.hardware.DcMotorSimple;
import com.qualcomm.robotcore.hardware.Servo;

import org.firstinspires.ftc.teamcode.libraries.PictographIdentifier;

public class RelicRecoveryRobot extends MecanumRobot {
    public DcMotor mWinchMotor;
    private Servo mJewelStick;

    private ColorSensor mColorSensor;

    public PictographIdentifier pictographIdentifier = new PictographIdentifier();

    public void setup(LinearOpMode linearOpMode) {
        super.setup(linearOpMode);

        mWinchMotor = linearOpMode.hardwareMap.dcMotor.get("winch motor");
        mWinchMotor.setDirection(DcMotorSimple.Direction.FORWARD);

        mJewelStick = linearOpMode.hardwareMap.servo.get("jewel stick");

        mColorSensor = linearOpMode.hardwareMap.colorSensor.get("color sensor");

        pictographIdentifier.activate();
    }

    public void setWinchPower(double power) {
        mWinchMotor.setPower(power);
    }

    public void setJewelStickPosition(double position) {
        mJewelStick.setPosition(position);
    }

    public void lowerJewelStick() {
        setJewelStickPosition(1.0);
    }

    public void raiseJewelStick() {
        setJewelStickPosition(0.0);
    }

    public ColorSensor getColorSensor() {
        return mColorSensor;
    }
}

File: ftc_app-master/TeamCode/src/main/java/org/firstinspires/ftc/teamcode/robots/RelicRecoveryRobot.kt
package org.firstinspires.ftc.teamcode.robots

import com.qualcomm.robotcore.eventloop.opmode.LinearOpMode
import com.qualcomm.robotcore.hardware.DcMotor
import com.qualcomm.robotcore.hardware.DcMotorSimple
import org.firstinspires.ftc.teamcode.librarys.PictographIdentifier

class RelicRecoveryRobot: MecanumRobot() {
    lateinit var mWinchMotor: DcMotor
    val pictographIdentifier = PictographIdentifier()

    override fun setup(linearOpMode: LinearOpMode) {
        super.setup(linearOpMode)

        mWinchMotor = linearOpMode.hardwareMap.dcMotor.get("winch motor")
        mWinchMotor.direction = DcMotorSimple.Direction.FORWARD

        pictographIdentifier.activate()
    }

    fun setWinchPower(power: Double) {
        mWinchMotor.power = power
    }
}